scipy/signal/_fir_filter_design.py

```diff
@@ -834,6 +835,248 @@ def remez(numtaps, bands, desired, *, weight=None, type='bandpass',
+    the specified design.
+
+    Parameters
+    ----------
+    freqs : array_like
+        A monotonic sequence of non-negative band edges in Hertz. All elements
+        must be less than half the sampling frequency ``fs``.
+    amps : array_like
+        A sequence containing the desired amplitudes of the signal to be
+        filtered over the corresponding frequency bands in ``freqs``.
+    rips : array_like
+        A sequence specifying the maximum allowable ripples of each band in the
+        frequency response.
+    fs : float, optional
+        The sampling frequency of the signal. Defaults to 1.0
+    alg : {'herrmann', 'kaiser', 'ichige'}, optional
+        Filter length approximation algorithm. Possible algorithms are
+        Herrmann [1]_, Kaiser [2]_, or Ichige [3]_. (Default: 'ichige')
+
+    Returns
+    -------
+    numtaps : int
+        The desired number of taps in the FIR filter.
+    bands : ndarray
+        A monotonic sequence containing the band edges.
+    desired : ndarray
+        A sequence half the size of bands containing the desired gain in each
+        of the specified bands.
+    weight : ndarray
+        A relative weighting to give to each band region.
+
+    Raises
+    ------
+    ValueError:
+        - If any element in ``freqs`` is negative, greater than 0.5, or the
+          length is not valid.
+        - If the lengths of ``amps`` and ``rips`` are not equal or one more
+          than half the length of ``freqs``.
+        - If any element in ``rips`` is negative.
+        - If ``alg`` is not a valid string option.
+
+    See Also
+    --------
+    remez
+
+    Notes
+    -----
+    .. versionadded:: 1.15.0
+
+    References
+    ----------
+    .. [1] O. Herrmann, L.R. Raviner, and D.S.K. Chan, Practical Design Rules
+           for Optimum Finite Impulse Response Low-Pass Digital Filters, Bell
+           Syst. Tech. Jour., 52(6):769-799, Jul./Aug. 1973.
+    .. [2] J.F. Kaiser, Nonrecursive Digital Filter Design Using I_0-sinh
+           Window function, Proc. IEEE Int. Symp. Circuits and Systems, 20-23,
+           April 1974.
+    .. [3] K. Ichige, M. Iwaki, and R. Ishii, Accurate Estimation of Minimum
+           Filter Length for Optimum FIR Digital Filters, IEEE Transactions on
+           Circuits and Systems, 47(10):1008-1017, October 2000.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.signal import remezord
+    >>> rp, rs = 3, 40  # Passband ripple, Stopband ripple
+    >>> fs = 2000  # Sampling frequency
+    >>> freqs = np.array([500, 600])  # Band edges
+    >>> amps = np.array([1, 0])  # Amplitudes
+    >>> rips = [(10**(rp/20)-1)/(10**(rp/20)+1), 10**(-rs/20)]  # Max ripples
+    >>> numtaps, bands, desired, weights = remezord(freqs, amps, rips, fs=fs)
+    >>> numtaps
+    27
+    >>> bands
+    [0, 0.25, 0.3, 0.5]
+    >>> desired
+    [1., 0.]
+    >>> weights
+    [1., 17.09973573]
```

Comment:
It would be nice to create the resulting filter and plot its frequency response using `freqz`... then you could at least eyeball that the results are consistent. Bonus points if you can make it look like Example 1 from the [linked website](http://www.ece.northwestern.edu/local-apps/matlabhelp/toolbox/signal/remezord.html)!

Reply:
I added the plot and it seems ok for me:

https://github.com/scipy/scipy/pull/20983/files#diff-c65308cc23ecff9cc1a62529e3298c1f26440a5fa95787fee881449ef8f1843bR1037-R1051

I also created a second example based on `remez` _pass-band_ filter example.